lib/services/ api_service.dart

```diff
@@ -3,33 +3,46 @@ import 'package:http/http.dart' as http;
 import '../models/ sunrise_sunset_model.dart';
 import '../models/lyrics_model.dart';
 
-class Resource<Item> {
-  final Item? data;
-  final String? error;
 
-  Resource.success(this.data) : error = null;
-  Resource.failure(this.error) : data = null;
+abstract class Resource<Item> {}
```

Comment:
Why is this an abstract class? Is Item a class? If so, remove that and pass generic type T